### PR TITLE
[MM-34838] Remove Ancillary Permissions That Are Purposefully Being Remove

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Adds a new endpoint to retrieve ancillary permissions for corresponding section permission.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-34838

Required for https://github.com/mattermost/mattermost-server/pull/17466

### CODE REVIEW

File: v4/source/permissions.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+  /permissions/ancillary_permissions:
```
Comment (hahmadia) [inline v4/source/permissions.yaml:1]:
```suggestion
  /permissions/ancillary:
```

**Review by mkraft — APPROVED** [review]

File: v4/source/permissions.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+  /permissions/ancillary:
+    get:
+      tags:
+        - permissions
+      summary: Return all system console subsection ancillary permissions
```
Comment (sbishel) [inline v4/source/permissions.yaml:5]: This actually returns all the system console subsection permissions appended to the parameters being passed in.

File: v4/source/permissions.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      description: >
+        Returns all the ancillary permissions for the corresponding system console
+        subsection permissions.
```
Comment (sbishel) [inline v4/source/permissions.yaml:8]:
```suggestion
        Returns all the ancillary permissions for the corresponding system console
        subsection permissions appended to the requested permission subsections.
```

File: v4/source/permissions.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+
+        __Minimum server version__: 5.35
+      parameters:
+        - name: subsection_permissions
+          in: query
+          description: >
+            The subsection permissions you want the ancillary permissions off.
```
Comment (sbishel) [inline v4/source/permissions.yaml:16]:
```suggestion
            The subsection permissions to return the ancillary permissions for.
```

File: v4/source/permissions.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            These values are comma seperated.
+            Ex. subsection_permissions=sysconsole_read_reporting_site_statistics,sysconsole_write_reporting_site_statistics,sysconsole_write_user_management_channels
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Successfully returned all ancillary permissions
```
Comment (sbishel) [inline v4/source/permissions.yaml:23]:
```suggestion
          description: Successfully returned all ancillary and requested permissions
```

**Review by sbishel — APPROVED** [review]

**Pull request merged**